source/classes.tex

```diff
@@ -6629,102 +6629,6 @@
 move construction from the object with automatic storage duration to \tcode{t2} that is elided.
 \end{example}
 
-\pnum
```

Comment:
FWIW, this refactoring merge-conflicts with the paper D2266 I've just started writing to clean up the semantics of implicit move. See https://quuxplusone.github.io/draft/d2266-implicit-move-rvalue-ref.html (which will at some point become https://wg21.link/p2266, I expect).

So basically I'm not in favor of any such major wording-moving-around, until we've finished figuring out what the wording should _say_. At least it'd be nice to avoid duplicating it in multiple places.

Reply:
My fundamental concern with the status quo is that mandatory behavior is hidden in a "copy elision" section, without any first-order connection to where the grammar appears (stmt.return etc.) If your paper would do the move, that's fine as well.

Reply:
Well, @zygoloid's review feedback on the above iteration of D2266 (R0 draft revision 1) is that _my_ wording change is totally in the wrong place and ought to be somewhere more like https://eel.is/c++draft/basic.lval . (However, I can't find any current wording that clearly states "an _id-expression_ is an lvalue"; if such wording existed, presumably that'd be the place for me to try changing.)

So I no longer object to shuffling _this_ wording around.

The new term of art "result-initialization" is intended to be useful for more than just implicit-move, right? Because if D2266 ends up changing the value-category of an implicitly-movable _id-expression_, then we won't need any wording to describe implicit-move anymore.

Reply:
This pull request (and the associated copy elision wording) will probably vanish entirely with your paper.

The wording that an /id-expression/ is an lvalue is in [expr.prim.id.unqual] and siblings.